Foundation/FileManager.swift

```diff
@@ -509,28 +509,172 @@ open class FileManager : NSObject {
+                defer {
+                    times.deinitialize(count: 2)
+                    times.deallocate()
+                }
+                if lutimes(pathPtr, times) < 0 {
+                    throw NSError(domain: NSPOSIXErrorDomain, code: Int(errno))
+                }
+            }
+        }
+        
+    }
+    
     /* setAttributes:ofItemAtPath:error: returns YES when the attributes specified in the 'attributes' dictionary are set successfully on the item specified by 'path'. If this method returns NO, a presentable NSError will be provided by-reference in the 'error' parameter. If no error is required, you may pass 'nil' for the error.
      
         This method replaces changeFileAttributes:atPath:.
      */
     open func setAttributes(_ attributes: [FileAttributeKey : Any], ofItemAtPath path: String) throws {
-        for attribute in attributes.keys {
-            if attribute == .posixPermissions {
-                guard let number = attributes[attribute] as? NSNumber else {
-                    fatalError("Can't set file permissions to \(attributes[attribute] as Any?)")
+        for (key, value) in attributes {
+            switch key {
+            case .creationDate:
+                guard let date = value as? Date else {
+                    fatalError("Can't set file modification date to \(value as Any?)")
+                }
+                try updateTimestamp(path: path, creation: date)
+                
+            case .modificationDate:
+                guard let date = value as? Date else {
+                    fatalError("Can't set file modification date to \(value as Any?)")
+                }
+                try updateTimestamp(path: path, modification: date)
+                
+            case .posixPermissions:
+                guard let number = value as? NSNumber else {
+                    fatalError("Can't set file permissions to \(value as Any?)")
                 }
                 #if os(macOS) || os(iOS)
-                    let modeT = number.uint16Value
+                let modeT = number.uint16Value
                 #elseif os(Linux) || os(Android) || CYGWIN
-                    let modeT = number.uint32Value
+                let modeT = number.uint32Value
                 #endif
                 _fileSystemRepresentation(withPath: path, {
                     if chmod($0, mode_t(modeT)) != 0 {
                         fatalError("errno \(errno)")
                     }
                 })
-            } else {
-                fatalError("Attribute type not implemented: \(attribute)")
+                
+            case .ownerAccountID:
+                guard let id = value as? NSNumber else {
+                    fatalError("Can't set file owner to \(value as Any?)")
+                }
+                _ = self._fileSystemRepresentation(withPath: path) {
+                    chown($0, uid_t(id.uint32Value), 0 &- 1)
+                }
```

Comment:
All of the `chmod` / `chown` system calls should throw with `errno` if the call fails

Reply:
You are right. I shall fix